src/main/scala/lms/core/stub.scala

```diff
@@ -24,6 +24,16 @@ object Adapter extends FrontEnd {
   var typeMap: mutable.Map[lms.core.Backend.Exp, Manifest[_]] = _
```

Comment:
minor and unrelated: does typeMap have to be `var` and mutable map at the same time?

Reply:
I think it has to be `var`. 
I think it is better to be `mutable.Map` so that it is faster.

Reply:
I think if `typeMap` is already a mutable map, then it doesn't have to be var. In that case
```
typeMap = new scala.collection.mutable.HashMap[lms.core.Backend.Exp, Manifest[_]]()
```
can be just `typeMap.clear()`.

Reply:
No :) That is exactly why it has to be var. We need 2 typeMap being alive at the same time (one old, one new)
if we do `typeMap.clear()`, we don't have the old one any more.

Reply:
well, we can clone the map :) let me get a PR on that.

Reply:
but is `clone` and `clear` more expensive than moving a var?

Reply:
I guess atomically it is, but I don't think here it plays a big role on performance. See the proposed changes in my PR #47. It also uses `transform` instead of `map` when checking if two canonical representations exist in funTable.

Reply:
reviewed that PR :)